server/src/main/java/org/cloudfoundry/identity/uaa/provider/IdentityProviderEndpoints.java

```diff
@@ -135,7 +135,12 @@ public IdentityProviderEndpoints(
     public ResponseEntity<IdentityProvider> updateIdentityProvider(@PathVariable String id, @RequestBody IdentityProvider body, @RequestParam(required = false, defaultValue = "false") boolean rawConfig) throws MetadataProviderException {
         body.setSerializeConfigRaw(rawConfig);
         String zoneId = identityZoneManager.getCurrentIdentityZoneId();
+
         IdentityProvider existing = identityProviderProvisioning.retrieve(id, zoneId);
+        if (existing == null) {
+            return new ResponseEntity<>(UNPROCESSABLE_ENTITY);
```

Comment:
Hmm, DELETE is also returning UNPROCESSABLE_ENTITY if the identity provider doesn't exist. Also, I have to update tests to look for the exception that is raised to return NOT_FOUND, rather than the response entity which is explicitly returned for UNPROCESSABLE_ENTITY. So I am going to make PUT/UPDATE return UNPROCESSABLE_ENTITY, and just GET return

Reply:
Agreed.  Let's be consistent.